### PR TITLE
Rename exception vectors

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

File: core/arch/arm/kernel/thread_private.h [hunk 1]
```diff
@@ -165,13 +165,15 @@ extern thread_pm_handler_t thread_system_reset_handler_ptr;
...
-void thread_vect_table(void);
-
 /*
  * Initializes VBAR for current CPU (called by thread_init_per_cpu()
  */
 void thread_init_vbar(void);
 
+void thread_excp_vect(void);
+void thread_excp_vect_workaround(void);
+void thread_excp_vect_workaround_a15(void);
```
Comment (lorc) [inline core/arch/arm/kernel/thread_private.h]: Actually, patch "core: rename exception vectors"  does two thing: renames vectors and adds global definitions for them.

I think you need to cover this in the commit message, or to do this in separate changes.

File: core/arch/arm/kernel/thread.c [hunk 2]
```diff
@@ -929,6 +929,49 @@ static void init_sec_mon(size_t pos __maybe_unused)
 #endif
 }
 
+static uint32_t __maybe_unused get_midr_implementer(uint32_t midr)
+{
+	return (midr >> MIDR_IMPLEMENTER_SHIFT) &
+	       (BIT(MIDR_IMPLEMENTER_WIDTH) - 1);
```
Comment (lorc) [inline core/arch/arm/kernel/thread.c]: It is a bit weird to see how you call `BIT()` macro with `_WIDTH` parameter. Maybe it is better to add proper definition for `MIDR_IMPLEMENTER_MASK` in header file?

File: core/arch/arm/kernel/thread.c [hunk 3]
```diff
@@ -929,6 +929,49 @@ static void init_sec_mon(size_t pos __maybe_unused)
...
+}
+
+static uint32_t __maybe_unused get_midr_primary_part(uint32_t midr)
+{
+	return (midr >> MIDR_PRIMARY_PART_NUM_SHIFT) &
+	       (BIT(MIDR_PRIMARY_PART_NUM_WIDTH) - 1);
```
Comment (lorc) [inline core/arch/arm/kernel/thread.c]: The same as with `BIT(MIDR_IMPLEMENTER_WIDTH)`

File: core/arch/arm/kernel/thread_private.h (hunk 1 above)
Comment (jenswi-linaro) [reply core/arch/arm/kernel/thread_private.h]: OK, I'll update the commit message

File: core/arch/arm/kernel/thread.c (hunk 2 above)
Comment (jenswi-linaro) [reply core/arch/arm/kernel/thread.c]: Yes, I'll do that.

**Comment by jenswi-linaro** [comment]
Update

**Comment by lorc** [comment]
`Reviewed-by: Volodymyr Babchuk <vlad.babchuk@gmail.com>`

**Comment by jenswi-linaro** [comment]
Thanks, squashed and tag applied

**Pull request merged**